src/Microsoft.TestPlatform.Utilities/InferRunSettingsHelper.cs

```diff
@@ -718,4 +718,84 @@ private static bool IsFrameworkIncompatible(Framework sourceFramework, Framework
         return !sourceFramework.Name.Equals(Framework.DefaultFramework.Name, StringComparison.OrdinalIgnoreCase)
                && !sourceFramework.Name.Equals(targetFramework.Name, StringComparison.OrdinalIgnoreCase);
     }
+
+    public static bool UpdateCollectCoverageSettings(XmlDocument xmlDocument)
+    {
+        var root = xmlDocument.DocumentElement;
+
+        var dataCollectorNodes = root?.SelectNodes("DataCollectionRunSettings/DataCollectors/DataCollector");
+        if (dataCollectorNodes == null)
+        {
+            return false;
+        }
```

Comment:
What does this mean?
If user is not providing runsettings explicitly but is running `dotnet test --collect "Code Coverage"`, what do we get here?

Is the `--collect` transformed automatically and we will get non-null here?